### PR TITLE
Combat Engi Rework; Electric Boogaloo Edition

### PR DESCRIPTION
**Pull request by Swagile**
## About The Pull Request

Revives #6099 with some extra additions.

Makes engi materials much less expensive to buy via points, gives a significant buff to engi starter, buffs construction pouch to 4 slots instead of 3 to compensate for all these extra mats.

## Why It's Good For The Game

Engi materials were always the worst thing about combat engi; you can barely buy any and your starter kit is trash. On top of that, your storage options suck for mats even if you somehow get more later on. This PR aims to solve those issues, which will allow combat engis to actually be combat engis; both shooting at the enemy, cading flanks, and setting up miners without having to ebeg if you try to do only one of these. Currently combat engis only gameplay is plasma cutter memes, or hugging a miner for 20 minutes in hopes of getting materials from a RO (and the round usually ends by then due to the current rush meta).

With this PR, engis will be able to build on a miner, and setup FOB's and cade off flanks with mats to spare.

The only thing I can see is an issue is mega FOB's, but most of these die due to not being guarded (with the only fringe case where mega FOB's matter is if marines camp the FOB forever like on Icey Caves but thats a map issue on top of another issue entirely), and is an integral issue to the game in of itself due to xenos bad siege; that's not something I can solve with this PR alone, sorry.

## Changelog
:cl:
balance: Construction pouches get 1 extra storage slots.
balance: Essential combat engi kit now gets 50 sandbags, 50 metal, 30 plasteel.
balance: Material points buffed; its much less expensive to use your points for materials in combat engi point vendor now.
/:cl:

### CODE REVIEW

**Comment by Swagile** [comment]
engi mains

GIVE ME YOUR ENERGY

File: code/game/objects/items/storage/pouch.dm [hunk 1]
```diff
@@ -621,7 +621,7 @@
 /obj/item/storage/pouch/construction
 	name = "construction pouch"
 	desc = "It's designed to hold construction materials - glass/metal sheets, metal rods, barbed wire, cable coil, and empty sandbags. It also has a hook for an entrenching tool."
-	storage_slots = 3
+	storage_slots = 5
```
Comment (BraveMole) [inline code/game/objects/items/storage/pouch.dm:624]: storage creep.

File: code/game/objects/machinery/vending/new_marine_vendors.dm [hunk 2]
```diff
@@ -422,7 +422,7 @@ GLOBAL_LIST_INIT(marine_selector_cats, list(
 		/obj/item/armor_module/storage/general = list(CAT_MOD, "General Purpose Storage Module", 0, "black"),
 		/obj/item/armor_module/storage/engineering = list(CAT_MOD, "Engineering Storage Module", 0, "black"),
 		/obj/item/storage/pouch/shotgun = list(CAT_POU, "Shotgun shell pouch", 0, "black"),
-		/obj/item/storage/pouch/construction = list(CAT_POU, "Construction pouch", 0, "orange"),
+		/obj/item/storage/pouch/construction/full = list(CAT_POU, "Construction pouch", 0, "orange"),
```
Comment (BraveMole) [inline code/game/objects/machinery/vending/new_marine_vendors.dm:425]: this plus the the price of mats in vendor is too much

**Review by BraveMole — CHANGES_REQUESTED** [review]

File: code/game/objects/items/storage/pouch.dm (hunk 1 above)
Comment (BraveMole) [reply code/game/objects/items/storage/pouch.dm:624]: You 're not supposed to hold 150000 kh of mats on you

**Comment by DerrickChamberlain** [comment]
Yes. Bigger Construction pouch, cheaper and faster mats. YES PLEASE! Along with allowing you to put filled sandbags in them this is a good PR push it through coder men!

File: code/game/objects/machinery/vending/new_marine_vendors.dm (hunk 2 above)
Comment (Swagile) [reply code/game/objects/machinery/vending/new_marine_vendors.dm:425]: so back to empty construction pouch?

File: code/game/objects/items/storage/pouch.dm (hunk 1 above)
Comment (Swagile) [reply code/game/objects/items/storage/pouch.dm:624]: don't see how this is an issue tho? you need this much space to hold the extra mats you will now get from this PR, and there aren't many options for engis except for jaeger engi (which forces you to go jaeger + slows you down)

File: code/game/objects/machinery/vending/new_marine_vendors.dm (hunk 2 above)
Comment (BraveMole) [reply code/game/objects/machinery/vending/new_marine_vendors.dm:425]: yes IMO

**Comment by Losenis** [comment]
"Storage creep" like it's not already balanced with an engi getting brazil'd with 200 points worth of mats

**Comment by DerrickChamberlain** [comment]
> "Storage creep" like it's not already balanced with an engi getting brazil'd with 200 points worth of mats

Even more of a reason to scream at them when they die to shame them. Good if you ask me.

**Comment by Swagile** [comment]
made a separate pull request to address the issue of storage creep

construction pouch getting 2 more spaces 

OR

combat backpack in return for a 15 point hit

either or i feel is needed to deal with the influx of mats this will add to a combat engis loadout

File: code/game/objects/machinery/vending/new_marine_vendors.dm (hunk 2 above)
Comment (Swagile) [reply code/game/objects/machinery/vending/new_marine_vendors.dm]: ok i put it back to empty variant

File: code/game/objects/items/storage/pouch.dm (hunk 1 above)
Comment (TiviPlus) [inline code/game/objects/items/storage/pouch.dm:624]: Make it 4

**Review by TiviPlus — CHANGES_REQUESTED** [review]

**Comment by TiviPlus** [comment]
Do not merge without TM
If engis abuse the new mats to just FOB harder this PR wont pass so I reccomend engis dont do that

File: code/game/objects/items/storage/pouch.dm (hunk 1 above)
Comment (DerrickChamberlain) [reply code/game/objects/items/storage/pouch.dm]: Four could be fine.

**Comment by Dolth** [comment]
Yo give detpack pouches.

**Pull request merged**